entry/entry.go

```diff
@@ -28,13 +29,88 @@ type Entry struct {
 	Clock    *LamportClock              `json:"clock,omitempty"`
 }
```

Comment:
Maybe you could make privates all attributes that are now accessibles through setters / getters below? (not sure if this is relevant)

Reply:
That makes sense, however that require a deeper refactor, we'll do that later